source/features/upload-button.js

```diff
@@ -3,22 +3,42 @@ import select from 'select-dom';
 import delegate from 'delegate';
 import {metaKey} from '../libs/utils';
 import * as icons from '../libs/icons';
+import observeEl from '../libs/simplified-element-observer';
 
-function addButtons() {
+let delegations;
+
+async function addButtons() {
 	for (const form of select.all('.js-previewable-comment-form:not(.rgh-has-upload-field)')) {
 		if (!select.exists('.js-manual-file-chooser[type=file]', form)) {
 			continue;
 		}
-		form.classList.add('rgh-has-upload-field');
-		const toolbarPosition = select('.js-saved-reply-container', form);
-		if (!toolbarPosition) {
-			continue;
-		}
-		toolbarPosition.after(
-			<button type="button" class="toolbar-item rgh-upload-btn">
-				{icons.cloudUpload()}
-			</button>
-		);
+
+		const toolbar = select('markdown-toolbar', form);
+
+		const observer = observeEl(toolbar, () => {
```

Comment:
The code inside this function should probably be what was before, it accomplishes the same in fewer lines:

```js

		form.classList.add('rgh-has-upload-field');
		const toolbarPosition = select('.js-saved-reply-container', form);
		if (!toolbarPosition) {
			continue;
		}
		toolbarPosition.after(
			<button type="button" class="toolbar-item rgh-upload-btn">
				{icons.cloudUpload()}
			</button>
		);
```

Reply:
Mutations can happen on multiple elements on the header element, we cannot assume that after the first mutation, the toolbar is ready.

So we keep checking, and only disconnect when the button is appended.

Reply:
True, I kept the check, but shortened the function in f688b2